Azure.MsBuild.Tasks/AzureDeploy.ps1

```diff
@@ -63,63 +63,65 @@ function WorkerRole([string]$SubscriptionName, [string]$Location, [string]$Worke
 			#Parameter Set: CommonSettings
 			#Set-AzureSubscription [-SubscriptionName] <String> [-Certificate <X509Certificate2> ] [-CurrentStorageAccountName <String> ] [-PassThru] [-ServiceEndpoint <String> ] [-SubscriptionDataFile <String> ] [-SubscriptionId <String> ] [ <CommonParameters>]
 
-			Set-AzureSubscription -SubscriptionName $SubscriptionName -CurrentStorageAccountName $WorkerRoleName
+			Select-AzureSubscription -SubscriptionName $SubscriptionName -Current
+			Set-AzureSubscription -SubscriptionName $SubscriptionName -CurrentStorageAccountName $WorkerRoleName 
 
```

Comment:
Set Subscription only ...Configures and stores common Windows Azure subscription settings.